weather/www/js/services.js

```diff
@@ -0,0 +1,214 @@
+/* Notes on forecast.io's API:
+ *  - daily.data[0] is today
+ * By: Cody Davey & Travis Pullen
+ */
+
+/* weatherStore holds data for weatherData service.
+ */
+weatherApp.factory('weatherStore', function() {
+    var weatherStore = {
+        current : {}
+    };
+
+    return weatherStore;
+});
+
+/* weatherData service gets weather data (current, forecasts, and
+ * historical), processes it as necessary, and provides it to controllers.
+ */
+weatherApp.service('weatherData', ['$q', '$resource', '$http',
+                                   'FORECASTIO_KEY', 'weatherStore',
+    function($q, $resource, $http, FORECASTIO_KEY, weatherStore) {
+        this.getCurrentWeather = function(lat, lng) {
+            var url = 'https://api.forecast.io/forecast/' +
+                FORECASTIO_KEY + "/" + lat + ',' + lng;
+
+            // JSONP is only needed for "ionic serve".
+            // Simpler $http.get(url) works on devices.
+            return $http.jsonp(url + '?callback=JSON_CALLBACK').then(
+                function success(resp) {
+                    weatherStore.current = resp.data;
+                    console.log('GOT CURRENT');
+                    console.dir(weatherStore.current);
+                },
+                function failure(error) {
+                    alert('Unable to get current conditions');
+                    console.error(error);
+                });
+        };
+
+        this.getLocation = function() {
+            return $q(function(resolve, reject) {
+                navigator.geolocation.getCurrentPosition(function(position) {
+                    resolve({
+                        latitude: position.coords.latitude,
+                        longitude: position.coords.longitude
+                    });
+                }, function(err) {
+                    reject(err);
+                });
+            });
+        };
+
+        // Round temp to tenths of a degree.
+        this.roundTemp = function(temp) {
+            if (Math.abs(temp) >= 10) {
+                return temp.toPrecision(3);
+            } else {
+                return temp.toPrecision(2);
+            }
+        };
+
+        // Return current temperature
+        this.tempNow = function() {
+            return this.roundTemp(weatherStore.current.currently.temperature);
+        };
+
```

Comment:
I think I mentioned these earlier before I saw that weatherDay2, 3, etc. were functions (thought they were arrays rather than functions that return arrays). It seems like the only difference between the functions is the index on data. It may be easier to have a general function that says

this.weather(index) {
    var temp = this.roundTemp(
    weatherStore.current.daily.data[index].temperatureMax);
    var precip = weatherStore.current.daily.data[index].precipIntensity;
    return [precip, temp,];
}

Your code works which is the bottom line -> just a quick tip that may help with expansion in the future.